src/main/java/frc/robot/subsystems/Arm.java

```diff
@@ -0,0 +1,56 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.*;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.lib.Constants;
+
+public class Arm extends SubsystemBase {
+    private final CANSparkMax armMotor;
+    private final CANSparkMax armFollowerMotor;
+
+    private final SparkPIDController armPID;
+
+    private final AbsoluteEncoder armEncoder;
+
+    public Arm() {
+        armMotor = new CANSparkMax(Constants.Arm.armMotor, CANSparkLowLevel.MotorType.kBrushless);
+        armPID = armMotor.getPIDController();
+        configArmMotor();
+
+        armFollowerMotor = new CANSparkMax(Constants.Arm.armFollowerMotor, CANSparkLowLevel.MotorType.kBrushless);
+        configFollowerMotor();
+
+        armEncoder = armMotor.getAbsoluteEncoder(SparkAbsoluteEncoder.Type.kDutyCycle);
+
+    }
+
+    public void setAngle(Rotation2d angle) {
+        armPID.setReference(angle.getDegrees(), CANSparkMax.ControlType.kPosition);
+    }
+
+    public boolean endConditionIntake() {
+        Rotation2d angle = new Rotation2d(Constants.Arm.desiredIntakeAngle);
+        return armEncoder.getPosition() > (angle.getDegrees() + 1) && armEncoder.getPosition() < (angle.getDegrees() - 1);
+    }
```

Comment:
This logic should be part of a Command, not part of the subsystem.

Not all constants need to go in a `Constants` file, it'd be totally fine to put the "desiredIntakeAngle" in this class, e.g.

```java
public class Arm extends Subsystem {
    /**
     * Arm position for picking up notes from the intake
     */
    private static final Rotation2d ARM_INTAKE_POSITION = new Rotation2d(...);

    /**
     * Arm position for shooting notes into the Speaker
     */
    private static final Rotation2d ARM_SHOOT_POSITION = new Rotation2d(...);

    /**
     * Arm position for dropping notes into the Amp
     */
    private static final Rotation2d ARM_DROP_POSITION = new Rotation2d(...);

    // ... everything else
}
```